components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultOIDCClaimsCallbackHandler.java

```diff
@@ -620,12 +640,42 @@ private Map<String, String> getUserClaimsInLocalDialect(String username,
                         null);
     }
 
+    private Map<String, String> getUserClaimsInLocalDialectWithUserID(String userId,
+                                                                      UserRealm realm,
+                                                                      List<String> claimURIList)
+            throws UserStoreException {
+
+        AbstractUserStoreManager userstore = (AbstractUserStoreManager) realm.getUserStoreManager();
+        if (userstore == null) {
+            throw new UserStoreException("Unable to retrieve UserStoreManager");
+        }
+        Map<String, String> userClaims =
+                userstore.getUserClaimValuesWithID(userId, claimURIList.toArray(new String[0]), null);
+        if (log.isDebugEnabled()) {
+            log.debug("User claims retrieved from user store: " + userClaims.size());
+        }
+        return userClaims;
+    }
+
+    private boolean isSubOrgLocalUserSwitchOrg(AuthenticatedUser authorizedUser) {
```

Comment:
```suggestion
    private boolean isOrganizationSsoUser(AuthenticatedUser authorizedUser) {
```
Lets add a method comment saying, check if the user has authenticated with organization SSO.

Reply:
Addressed with c202638bd7d45d70c3c4030d1f697325e8fdbc4f